core/src/test/java/ai/timefold/solver/core/testutil/PlannerTestUtils.java

```diff
@@ -70,11 +70,12 @@ public static <Solution_> SolverConfig buildSolverConfig(Class<Solution_> soluti
                                 new TerminationConfig().withStepCountLimit(TERMINATION_STEP_COUNT_LIMIT)));
     }
 
-    public static <Solution_> Solution_ solve(SolverConfig solverConfig, Solution_ problem) {
+    public static synchronized <Solution_> Solution_ solve(SolverConfig solverConfig, Solution_ problem) {
         return solve(solverConfig, problem, true);
     }
 
-    public static <Solution_> Solution_ solve(SolverConfig solverConfig, Solution_ problem, boolean bestSolutionEventExists) {
+    public static synchronized <Solution_> Solution_ solve(SolverConfig solverConfig, Solution_ problem,
```

Comment:
I'd add a comment explaining the synchronized keyword; we're doing it because metrics are shared mutable state.